src/rust/ide/view/graph-editor/src/component/node.rs

```diff
@@ -689,21 +686,25 @@ impl Node {
             eval bg_color ((c) model.background.bg_color.set(c.into()));
 
 
-            // === Tooltip ===
+            // === Output Label ===
 
-            // Hide tooltip if we show the preview vis.
-            frp.source.tooltip <+ preview_visible.on_true().constant(tooltip::Style::unset_label());
-            // Propagate output tooltip. Only if it is not hidden, or to disable it.
-            block_tooltip      <- hide_tooltip && has_tooltip;
-            frp.source.tooltip <+ model.output.frp.tooltip.gate_not(&block_tooltip);
+            model.output.set_port_label_visibility
+                <+ visualization_visible.not().and(&no_error_set);
 
 
             // === VCS Handling ===
+
             model.vcs_indicator.frp.set_status <+ frp.set_vcs_status;
         }
-
+        
         model.error_visualization.set_layer(visualization::Layer::Front);
-        frp.set_error.emit(None);
+
+        // Those three lines initialize the values that are necessary for port labels to show.
+        // All three are needed in this order, for reasons that I do not understand.
+        frp.set_error(None);
+        frp.enable_visualization();
+        frp.disable_visualization();
```

Comment:
Few things here:

1. Do not leave such comments in the code, please. This is a PR, not a Draft PR. It means that you propose these changes. So you propose that we add to code a comment "All three are needed in this order, for reasons that I do not understand.". Until this issue is resolved, this should be marked as a Draft PR, not a real PR.
2. `frp.enable_visualization();frp.disable_visualization();` looks bad. Take a look at Adam's explanation above and design the FRP netwrok initialization based on what Adam has described.